### PR TITLE
Introduce an unused annotation

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes scala/bug#10790

### CODE REVIEW

File: src/library/scala/annotation/unused.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/** An annotation that designates that the target is unused. */
+final class unused extends deprecated("unused", "")
```
Comment (He-Pin) [inline src/library/scala/annotation/unused.scala:23]: Are you sure it's a good name, I can live with it.

Comment (dwijnand) [reply src/library/scala/annotation/unused.scala:23]: Yeah, I'm happy with it.

Comment (hepin1989) [reply src/library/scala/annotation/unused.scala]: great then。

File: src/library/scala/annotation/unused.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/** An annotation that designates that the target is unused. */
```
Comment (lrytz) [inline src/library/scala/annotation/unused.scala]: The doc should say
  - why this extends `deprecated` and is not unrelated
  - what the compiler does with declarations annotated `@unused`
  - what it is recommended to be used for

**Comment by dwijnand** [comment]
Let's try the low level approach.

**Comment by SethTisue** [comment]
>  * This extends `@deprecated` because the compiler's unused linting ignores items annotated with
> * `@deprecated`.

@som-snytt would it be difficult for the unused-detection code to recognize `@unused` directly? otherwise it seems like piling hack upon hack. annotations extending each other seems weird to me, and hijacking `@deprecated` in this context was a hack in the first place

**Comment by som-snytt** [comment]
Although this solution has a certain elegance to it, superseded by https://github.com/scala/scala/pull/7623 which instead of piling hack upon hack, just shoves it off to one side and maybe spreads it around a little to make sure the garden is evenly fertilized.

**Pull request closed without merging**